Automattic/jetpack
# Identity Crisis: delete the migrate_for_idc option when a migrated_for_idc response is received

#### Changes proposed in this Pull Request:
* Delete the `migrate_for_idc` option when a `migrated_for_idc` response is received. WPCOM will migrate to the new URLS when the `migrate_for_idc` query parameter is received with a remote request. After migrating, WPCOM will add `migrated_for_idc` to the response body. Check for that response and delete the `migrate_for_idc` option when it has been received.

#### Jetpack product discussion
* p9dueE-3oW-p2

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
You can test by adding the `migrated_for_idc` field to the REST request response using a filter. For example,

1. Install and activate this branch of Jetpack. Connect Jetpack.
2. Add this snippet to your site. You can use the Code Snippets plugin if you'd like.
```
add_filter( 'http_response', 'add_migrate_for_idc' );
function add_migrate_for_idc( $response ) {
	$response_body = json_decode( wp_remote_retrieve_body( $response ), true );
	$response_body['migrated_for_idc'] = true;
	$response['body'] = json_encode( $response_body );
	return $response;
}
```
3. Set the `migrate_for_idc` option: `wp option update jetpack_migrate_for_idc 1`
4. Do something that causes a remote request. For example, go to a wp-admin page.
5. Verify that the `migrate_for_idc` option was emptied: `wp option get jetpack_migrate_for_idc`
